css/selectors/invalidation/user-action-pseudo-classes-in-has.html

```diff
@@ -37,47 +38,56 @@
 </div>
 <script>
   const tab_key = '\ue004';
-  const hovermeRect = hoverme.getBoundingClientRect();
-  const focusme1Rect = focusme1.getBoundingClientRect();
 
   promise_test(async () => {
     assert_equals(getComputedStyle(subject1).color, "rgb(0, 0, 0)", "subject1 initially black");
     assert_equals(getComputedStyle(subject2).color, "rgb(0, 0, 0)", "subject3 initially black");
 
     await new test_driver
-      .Actions()
-      .pointerMove(hovermeRect.x + 1, hovermeRect.y + 1, {origin: "viewport"})
-      .send();
+        .Actions()
+        .pointerMove(0, 0, {origin: hoverme})
+        .send();
     assert_equals(getComputedStyle(subject1).color, "rgb(0, 0, 255)",
                   "subject1 should be blue");
     assert_equals(getComputedStyle(subject3).color, "rgb(0, 0, 128)",
                   "subject3 should be navy");
 
     await new test_driver
-      .Actions()
-      .pointerMove(hovermeRect.x + 1, hovermeRect.y + 1, {origin: "viewport"})
-      .pointerDown()
-      .send();
+        .Actions()
+        .pointerMove(0, 0, {origin: unhoverme})
+        .send();
+    assert_equals(getComputedStyle(subject1).color, "rgb(0, 0, 0)",
+                  "subject1 should be back to black");
+    assert_equals(getComputedStyle(subject3).color, "rgb(0, 0, 0)",
+                  "subject3 should be back to black");
+
+
+    await new test_driver
+        .Actions()
+        .pointerMove(0, 0, {origin: hoverme})
+        .pointerDown()
+        .send();
     assert_equals(getComputedStyle(subject1).color, "rgb(135, 206, 235)",
                   "subject1 should be skyblue");
     assert_equals(getComputedStyle(subject3).color, "rgb(173, 216, 230)",
                   "subject3 should be lightblue");
 
     await new test_driver
-      .Actions()
-      .pointerUp()
-      .send();
-    assert_equals(getComputedStyle(subject1).color, "rgb(0, 0, 255)",
-                  "subject1 should be blue again");
-    assert_equals(getComputedStyle(subject3).color, "rgb(0, 0, 128)",
-                  "subject3 should be navy again");
+        .Actions()
+        .pointerUp()
+        .pointerMove(0, 0, {origin: unhoverme})
```

Comment:
This removes coverage for what was previously tested, which was to just release the mouse button without moving it. Can you keep this part of the test?

```suggestion
        .pointerMove(0, 0, {origin: hoverme})
```

And keep the assertions as they were. If you think it's also useful to test moving the mouse right after mouseup, additional steps for that would be OK, but I'm not sure I see the difference to the first unhoverme bits.

Reply:
It does remove that coverage - but does the suggested change really test "going from `:hover:active` to `:hover`?"
I could be misreading this, but the current & suggested action sequence decomposes into two ticks, right? So it'd be a succession of "`:has(:hover:active)` invalidates" (Which is similar to the current change), then "`:has(:hover)` invalidates" (Which is already tested)?